### ISSUE TITLE
Tweak layout of Alarm Panel card editor

### ISSUE THREAD

**Issue by SeanPM5** [home-assistant/home-assistant-polymer#3675]
**Home Assistant release with the issue:**
0.98.5

**Description of problem:**
While reporting the preview bug in https://github.com/home-assistant/home-assistant-polymer/issues/3662 I played around with this card editor a bunch. And I think that it could use a few layout and usability tweaks. It's mostly three minor suggestions:

1. The "Entity" dropdown should be on its own line. This is the only required part, yet it's currently displaying an optional field before it on the same line. Because it has such small width and because the `alarm_control_panel` domain is pretty lengthy, it's a very poor experience on mobile and even desktop.

2. With the "Name" textfield moved underneath, the placeholder text should also indicate that it is an optional field.

3. Switch to toggles instead of a dropdown. There's only four choices, and it already shows two by default, so having a dropdown for two items seems unnecessary. And since a dropdown is collapsed by default those extra options are also not immediately visible to the user. And a dropdown also requires extra clicks/taps compared to the toggles.

Here is a before and after mockup.

Current design:
![current-alarm-panel](https://i.imgur.com/Rm5Zq5l.png)

Proposed design:
![proposed-alarm-panel](https://i.imgur.com/PWoZY4w.jpg)

Now that you can compare both images, to recap, benefits of proposed design:
* Clearly labeled heading (as per https://github.com/home-assistant/home-assistant-polymer/issues/3658)
* Entity ID being the first option shows it has priority/significance over everything else
* Can actually read the entire entity ID now
* Name field is clearly labeled as optional so there's no confusion and user can easily skip it
* All of the "Available States" are immediately visible and no longer behind a dropdown
* Because they are toggles instead of dropdown choices you save clicks
* There's not a bunch of empty wasted space on the left
* A lot more mobile friendly

**Comment by zsarnett** [comment]
I think instead of the switches we should implement Chips
![image](https://user-images.githubusercontent.com/18730868/77119512-fcb39200-6a0c-11ea-9ad3-0afe2616274d.png)

**Comment by bramkragten** [comment]
That would be filter chips then right?
![image](https://user-images.githubusercontent.com/5662298/77158107-eeb05080-6aa2-11ea-9d59-8ccd20e0a42a.png)

Btw, we do have a chips component <https://github.com/home-assistant/frontend/blob/dev/src/components/ha-chips.ts>

**Comment by SeanPM5** [comment]
Some quick mockups. Just realized I forgot the theme dropdown, but whatever, this should give a general idea.

Toggle switches

![image](https://i.imgur.com/TNi62Cw.jpg)

Chips

![image](https://i.imgur.com/1VWpqjT.jpg)

Multiselect checkboxes

![image](https://i.imgur.com/33PzVjT.jpg)

I lean towards preferring toggle switches overall. They look the cleanest IMO and will be the most familiar for users as they're used in all of the other card editors. They can line up nicely and have adequate spacing, and they will probably scale down nicest on mobile.

Chips also look nice but they're not _quite_ as clean IMO. The different colors sticks out a bit too much for my personal taste. And I don't think it's as immediately obvious how chips are supposed to be interacted with, so you'd need heading text unlike with the toggles. My main thing though, it feels a little weird having oval-shaped filled buttons and rectangular-shaped outlined buttons right next to each other. The styles clash too much. But perhaps that's being too nitpicky on my part. I'd be fine with chips, they're just not my personal preference.

Multiselect is my least favorite, doesn't feel like the most efficient use of space. There's lots of empty unused space on the right side if you display them vertically. They could be displayed horizontally I suppose, but that feels a bit weird with checkboxes.

**Comment by zsarnett** [comment]
Agreed. I was hoping to create an email like experience when adding contacts to the `To` field. But to do that with Material Design is not the easiest. 

my suggestion would not be like filter chips. It would be like input chips.
![image](https://user-images.githubusercontent.com/18730868/77197157-0c89af80-6abb-11ea-8efb-d0069618cc4c.png)

**Comment by bramkragten** [comment]
We should do the switch version Sean started with.
